compiler/catala_utils/message.ml

```diff
@@ -404,3 +405,31 @@ let result = make ~level:Result ~cont:emit
 let results r = emit (List.flatten (List.map of_result r)) Result
 let warning = make ~level:Warning ~cont:emit
 let error = make ~level:Error ~cont:(fun m _ -> raise (CompilerError m))
+
+(* Multiple errors handling *)
+let global_errors = ref None
+
+let delayed_error x =
+  make ~level:Error ~cont:(fun m _ ->
+      match !global_errors with
+      | None ->
+        failwith
+          "delayed error called outside scope: encapsulate using \
+           'with_delayed_errors' first"
```

Comment:
Ideally this would be `error ~internal:true`: we'd like to avoid rogue `Failure`s

(also, including the original error might be helpful ? Not sure)